packages/quick_actions/quick_actions_android/android/src/main/java/io/flutter/plugins/quickactions/QuickActions.java

```diff
@@ -0,0 +1,183 @@
+import android.content.res.Resources;
+import android.graphics.drawable.Icon;
+import android.os.Build;
+import androidx.annotation.ChecksSdkIntAtLeast;
+import androidx.annotation.NonNull;
+import io.flutter.plugins.quickactions.Messages.AndroidQuickActionsApi;
+import io.flutter.plugins.quickactions.Messages.FlutterError;
+import io.flutter.plugins.quickactions.Messages.Result;
+import io.flutter.plugins.quickactions.Messages.ShortcutItemMessage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+final class QuickActions implements AndroidQuickActionsApi {
+  protected static final String EXTRA_ACTION = "some unique action key";
+
+  private final Context context;
+  private Activity activity;
+
+  QuickActions(Context context) {
+    this.context = context;
+  }
+
+  void setActivity(Activity activity) {
+    this.activity = activity;
+  }
+
+  public Activity getActivity() {
+    return this.activity;
+  }
+
+  @ChecksSdkIntAtLeast(api = Build.VERSION_CODES.N_MR1)
+  boolean isVersionAllowed() {
+    // We already know that this functionality does not work for anything
+    // lower than API 25 so we chose not to return error. Instead we do nothing.
+    return Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1;
+  }
+
+  @Override
+  public void setShortcutItems(
+      @NonNull List<ShortcutItemMessage> itemsList, @NonNull Result<Void> result) {
+    if (isVersionAllowed()) {
+      ShortcutManager shortcutManager =
+          (ShortcutManager) context.getSystemService(Context.SHORTCUT_SERVICE);
+      List<ShortcutInfo> shortcuts = ShortcutItemMessageToShortcutInfo(itemsList);
+      ThreadPoolExecutor executor =
+          new ThreadPoolExecutor(0, 1, 1, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
+
+      executor.execute(
+          () -> {
+            boolean dynamicShortcutsSet = false;
+            try {
+              shortcutManager.setDynamicShortcuts(shortcuts);
+              dynamicShortcutsSet = true;
+            } catch (Exception e) {
+              // Leave dynamicShortcutsSet as false
+            }
+
+            final boolean didSucceed = dynamicShortcutsSet;
+
+            new Thread(
+                    new Runnable() {
+                      @Override
+                      public void run() {
+                        if (didSucceed) {
+                          result.success(null);
+                        } else {
+                          result.error(
+                              new FlutterError(
+                                  "quick_action_setshortcutitems_failure",
+                                  "Exception thrown when setting dynamic shortcuts",
+                                  null));
+                        }
+                      }
+                    })
+                .start();
+          });
+    }
```

Comment:
You have a codepath that never calls `result`, which is an error.

Reply:
Seems like we should add unit tests that the no-op for old versions actually returns.

Reply:
This method is inherently asynchronous due to the thread bounce, so it still needs to be declared as async. The version you have now will return success immediately no matter what, and the later `throw` would not work correctly (maybe crash the app? I forget how unhandled throws work in Java).

Since the thread bounce here is [to keep `setDynamicShortcuts` off of the main thread](https://github.com/flutter/plugins/pull/4502#discussion_r750511289) we could simplify this a lot by removing both of them and using background channels for the plugin (the linked discussion saying it would break the platform interface is no longer true due to the change to in-package method channels, so it should be easy), but in the spirit of minimal changes I would make that part of the TODO below (i.e., change it to "TODO(camsim99): Investigate removing all of the executor logic in favor of background channels.") and keep the current implementation for now, with this method still `async`.

Reply:
Is this still an issue? I'm not certain where this was a problem originally, as I force pushed over the original code. It seems to me that all paths reply currently.

Reply:
oh, this was part of the change that I reverted already.